### PR TITLE
[wasm] WebSocket perf test

### PR DESCRIPTION
**Pull request by pavelsavara**
Performance of buffering operations

### CODE REVIEW

**Review by radekdoulik — APPROVED** [review]
LGTM

**Comment by pavelsavara** [comment]
The build failure is https://github.com/dotnet/runtime/issues/58812

File: src/mono/sample/wasm/browser-bench/WebSocket.cs [hunk 1]
```diff
@@ -120,51 +149,91 @@ public override void RunStep()
...
+
+            public override void RunStep()
+            {
+                if (step == 0)
+                {
+                    // this is GC step
+                    client.ReceiveAsync(new ArraySegment<byte>(new byte[1]), CancellationToken.None);
+                    return;
+                }
+                Console.WriteLine("RunStep " + step);
```
Comment (radekdoulik) [inline src/mono/sample/wasm/browser-bench/WebSocket.cs:221]: Is this print intended or forgotten debug?

File: src/mono/sample/wasm/browser-bench/BenchTask.cs [hunk 2]
```diff
@@ -56,31 +57,41 @@ protected virtual int CalculateSteps(int milliseconds, TimeSpan initTs)
 
         public Result RunBatch(BenchTask task, int milliseconds)
         {
-            DateTime start;
+            DateTime start = DateTime.Now;
```
Comment (radekdoulik) [inline src/mono/sample/wasm/browser-bench/BenchTask.cs:60]: Why do we want to set it here?

Comment (radekdoulik) [reply src/mono/sample/wasm/browser-bench/BenchTask.cs:60]: nm, that's for the case when an exception happens

**Review by radekdoulik — APPROVED** [review]

**Pull request merged**